src/Microsoft.Security.Utilities.Core/DataClassification/DAT101_000_GuidValue.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Security.Utilities
+{
+    public class GuidValue : RegexPattern
+    {
+        public GuidValue()
+        {
+            Id = "DAT101/000";
+            Name = nameof(GuidValue);
+            Pattern = @"(?i)^([0-9a-f]{8}-[0-9a-f]{4}-[0|9|a|b|c|d|e|f][0-9a-f]{3}-[0-9a-f]{4}-[0-9a-f]{12})" +
+                       "|([0-9a-f]{8}-[0-9a-f]{4}-[1-8][0-9a-f]{3}-[^89ab][0-9a-f]{3}-[0-9a-f]{12})$";
+        }
+
+        public override Tuple<string, string> GetMatchIdAndName(string match)
+        {
+            if (UuidValue.c_Nil.Equals(match, StringComparison.OrdinalIgnoreCase) || UuidValue.c_Max.Equals(match, StringComparison.OrdinalIgnoreCase))
+            {
+                return null;
+            }
+
+            return base.GetMatchIdAndName(match);
+        }
+
+        public override IEnumerable<string> GenerateTruePositiveExamples()
+        {
+            // Example of a GUID, but not a UUID. This example is the Office 365 SharePoint Online App Id. See https://learn.microsoft.com/en-us/troubleshoot/entra/entra-id/governance/verify-first-party-apps-sign-in.
+            yield return "00000003-0000-0ff1-ce00-000000000000";
+
+            // Example of almost a UUIDv4, but the first 'c' of the 4th block ('cbac') makes it not a UUID, but a GUID.
```

Comment:
>cbac [](http://example.com/codeflow?start=80&length=4)

ah, so this little wrinkle really point to the nuance you mentioned. but honestly, if GUIDs and UUIDs are all merged together, how the heck are you supposed to distinguish 'proper' UUIDs from, say, a predefined GUID where someone has chosen to share a common value for this reserved bit?